libutils/string_lib.c

```diff
@@ -1014,3 +1026,24 @@ bool StringAppendDelimited(char *dst, size_t *dst_len, size_t dst_size,
 
     return true;
 }
+
+/**
+ *  Append #sep if not already there, and then append #leaf.
+ */
+bool PathAppend(char *path, size_t path_size, const char *leaf, char sep)
+{
+    size_t path_len = strlen(path);
+    size_t leaf_len = strlen(leaf);
+
+    if (path_len + 1 + leaf_len >= path_size)
+    {
+        /* Log(LOG_LEVEL_ERR, "Internal limit reached in JoinPath(\"%s\", \"%s\")", */
+        /*     path, leaf); */
```

Comment:
Maybe remove this? Doesn't seem that relevant anymore, and it's easy to understand what's going on here anyway.

(Reposting in PR instead of commit)